mkosi/__init__.py

```diff
@@ -1637,6 +1639,37 @@ def bump_image_version(uid: int = -1, gid: int = -1) -> None:
     os.chown("mkosi.version", uid, gid)
 
 
+def show_docs(args: MkosiArgs) -> None:
+    if args.doc_format == DocFormat.auto:
+        formats = [DocFormat.system, DocFormat.markdown, DocFormat.pandoc, DocFormat.man]
+    else:
+        formats = [args.doc_format]
+
+    while formats:
+        form = formats.pop()
+        try:
+            if form == DocFormat.man:
+                with importlib.resources.path("mkosi.resources", "mkosi.1") as man:
+                    if not man.exists():
+                        raise FileNotFoundError
+                    return run(["man", "--local-file", man])
+            elif form == DocFormat.pandoc:
+                if not shutil.which("pandoc"):
+                    logging.warning("pandoc is not available")
+                with importlib.resources.path("mkosi.resources", "mkosi.md") as md:
+                    pandoc = run(["pandoc", "-t", "man", "-s", md], stdout=subprocess.PIPE)
+                    return run(["man", "--local-file", "-"], input=pandoc.stdout)
+            elif form == DocFormat.markdown:
+                md = importlib.resources.read_text("mkosi.resources", "mkosi.md")
+                return page(md, args.pager)
+            elif form == DocFormat.system:
+                return run(["man", "mkosi"])
+        except (FileNotFoundError, subprocess.CalledProcessError) as e:
```

Comment:
I would translate `FileNotFoundError` into a `die()` if we're going to raise so we don't show an ugly exception stacktrace.